### ISSUE TITLE
circular dependencies in message types

### ISSUE THREAD

**Issue by damiendr** [JuliaIO/ProtoBuf.jl#61]
I'm trying to compile a .proto file that contains the following structure:

```
message A {
  ...
  repeated B xxx = 9;
}
message B {
  ...
  optional A yyy = 4;
}
```

This fails with the following error:

```
--julia_out: All types could not be fully resolved:
mymessages.A depends on mymessages.B
mymessages.B depends on mymessages.A
```

Everything compiles fine with `--cpp_out`.

**Comment by damiendr** [comment]
Note: it seems that the problem will sooner or later involve a Julia language issue: https://github.com/JuliaLang/julia/issues/269

**Comment by BugDiver** [comment]
@damiendr  +1

**Comment by bar4ka** [comment]
Tried to use this on a real world scenario: https://github.com/bar4ka/POGOProtos
Compiled to end without errors but unusable at runtime with LoadError: LoadError: UndefVarError: 
Circular dependencies between modules are the cause

**Comment by tanmaykm** [comment]
fixed by #100
